File: db4oj.tests/src/com/db4o/db4ounit/common/cs/PrefetchObjectCountZeroTestCase.java
/* Copyright (C) 2009  Versant Inc.   http://www.db4o.com */
package com.db4o.db4ounit.common.cs;

import com.db4o.config.*;

import db4ounit.*;
import db4ounit.extensions.*;
import db4ounit.extensions.fixtures.*;

public class PrefetchObjectCountZeroTestCase extends AbstractDb4oTestCase implements OptOutAllButNetworkingCS {

	public static class Item {
	}
	
	@Override
	protected void configure(Configuration config) throws Exception {
		config.clientServer().prefetchObjectCount(0);
	}

	@Override
	protected void store() throws Exception {
		store(new Item());
	}

	public void testZeroPrefetchObjectCount() {
		Assert.isNotNull(retrieveOnlyInstance(Item.class));
	}
}

File: db4o.net/Db4objects.Db4o.Tests/Db4objects.Db4o.Tests/Common/CS/PrefetchObjectCountZeroTestCase.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

#if !SILVERLIGHT
using Db4oUnit;
using Db4oUnit.Extensions;
using Db4oUnit.Extensions.Fixtures;
using Db4objects.Db4o.Config;
using Db4objects.Db4o.Tests.Common.CS;

namespace Db4objects.Db4o.Tests.Common.CS
{
	public class PrefetchObjectCountZeroTestCase : AbstractDb4oTestCase, IOptOutAllButNetworkingCS
	{
		public class Item
		{
		}

		/// <exception cref="System.Exception"></exception>
		protected override void Configure(IConfiguration config)
		{
			config.ClientServer().PrefetchObjectCount(0);
		}

		/// <exception cref="System.Exception"></exception>
		protected override void Store()
		{
			Store(new PrefetchObjectCountZeroTestCase.Item());
		}

		public virtual void TestZeroPrefetchObjectCount()
		{
			Assert.IsNotNull(((PrefetchObjectCountZeroTestCase.Item)RetrieveOnlyInstance(typeof(
				PrefetchObjectCountZeroTestCase.Item))));
		}
	}
}
#endif // !SILVERLIGHT
